### ISSUE TITLE
Dependência de System.Web

### ISSUE THREAD

**Issue by rafakwolf** [BoletoNet/boletonet#8]
Em projetos windows forms está sendo obrigatória a dependência de System.Web. 
Talvez fosse interessante isolar a renderização dos boletos, dessa forma o uso de boletos html seria facultativa, logo não seria obrigatório o uso da System.Web.

**Comment by carloscds** [comment]
Nao vejo problemas em ter a System.Web. Mas qual seria sua sugestao neste caso ?

**Comment by stivencamara** [comment]
Podemos criar um projeto para o HTML. Assim desaclopamos a System .web.

Vou pensar nisso.

Stiven

Enviado do meu smartphone Sony Xperia™

---- Carlos dos Santos escreveu ----

Nao vejo problemas em ter a System.Web. Mas qual seria sua sugestao neste caso ?

---

**Comment by rafakwolf** [comment]
Exato, dessa forma a biblioteca poderia ser usada para cálculos e identificação de boleto apenas, no caso de uso com uma aplicação windows forms.

Realmente ter "System.Web" não é um problema, mas neste caso, é desnecessário.

Parabéns pelo trabalho de vocês.

**Comment by AlbertoMonteiro** [comment]
Acho que seria uma boa, mas no projeto para geração do html, eu trocaria essa dependência do System.Web pela dependência do Razor, a fim de ter um pipeline mais flexível na customização do html do boleto gerado.

**Comment by rafakwolf** [comment]
Olá pessoal, algum avanço em relação a isto?

O projeto da empresa que trabalho é Windows Forms, por isso sugeri este desacoplamento, a ideia seria fazer uma espécie de "core" do Boleto.NET para a biblioteca não ser específica para web e sim para qualquer tipo de projeto.

**Comment by carloscds** [comment]
@rafakwolf O que temos feito em projetos Windows é adicionar a System.Web. Para remover completamente acho que o impacto seria maior.

**Comment by rafakwolf** [comment]
@carloscds ok, podemos deixar esta issue em aberto para futuras sugestões no projeto ?

Obrigado

**Comment by carloscds** [comment]
@rafakwolf Sem problemas!

**Comment by renattomachado** [comment]
Penso que será interessante fazer o desacoplamento, pois com aplicações Asp.Net 5 (vNext) somos obrigados a incluir a System.Web. Creio que trocar pelo Razor será uma boa opção.

**Comment by rafaelsgripp** [comment]
Estou começando a utilizar a ferramenta, e estou precisando utilizar em uma aplicação windows forms, qual é a melhor maneira de utilizar a biblioteca nesse caso? Qual método devo chamar? Alguém tem um exemplo?

Desde já agradeço a ajuda

**Comment by AlbertoMonteiro** [comment]
@rafaelsgripp Você pode ver como usar através dos testes, segue os testes do **Banco do Brasil**

https://github.com/BoletoNet/boletonet/blob/master/src/Boleto.Net.Testes/BancoBrasil/BancoBrasilCarteira18Teste.cs

**Comment by cezarant** [comment]
@rafakwolf  tudo bem? Me parece que o assunto foi encerrado. Você poderia fechar a issue?

**Comment by carloscds** [comment]
@cezarant Na verdade não, o projeto continua com o System.Web.
